py-polars/src/series/construction.rs

```diff
@@ -233,8 +233,13 @@ impl PySeries {
     }
 
     #[staticmethod]
-    fn repeat(name: &str, val: Wrap<AnyValue>, n: usize, dtype: Wrap<DataType>) -> PyResult<Self> {
-        let av = val.0;
+    fn repeat(
+        value: Wrap<AnyValue>,
+        n: usize,
+        name: &str,
+        dtype: Wrap<DataType>,
+    ) -> PyResult<Self> {
+        let av = value.0;
         Ok(Series::new(name, &[av])
```

Comment:
If we instead of casting the `Series` which has length `n`, we cast the anyvalue `av` which is only one element. We save `O(n)`.

Reply:
That's exactly what I thought. But how would you do that? Not very familiar with the idea of anyvalues.. I guess you'd have to match the dtype and extract accordingly? Then we won't need the afterwards cast.

Reply:
It should have a cast method. `AnyValue::cast`

Reply:
See: https://github.com/pola-rs/polars/blob/5e0eade315eed519d399a4a81dcab44cfea9b219/polars/polars-core/src/datatypes/any_value.rs#L436

Reply:
That cast method is incomplete - for example, casting I cannot simply cast to the given dtype as the method does not include casting to itself (e.g. Boolean to Boolean, Utf8 to Utf8, ...).

So while this is a nice improvement, I would consider it out of scope for this PR.

Reply:
An anyvalue can convert a DataType, so we can check the DataTypes and if they are not equal, cast the anyvalue.

Reply:
Ok sure, can do that!

Reply:
Actually, if I read the code correctly here, it creates a Series with length 1, casts it, and then extends the Series. So casting the AnyValue first does not save O(n) at all. Or am I mistaken?

Reply:
Nope, You are right. Both cases are `O(1)`. Sorry for the confusion.